syndicate/silos/dev.py

```diff
@@ -0,0 +1,88 @@
+from syndicate.utils import action_log_group, action_log, action_error, fronted, silo_id_for
+import requests
+import pprint
+
+SILO_NAME = 'DEV'
+@action_log_group(SILO_NAME)
+def syndicate(posts, api_key):
+    """
+    Syndicates the given posts to https://dev.to, updating the ones that
+    already exist there and creating articles for the ones that don't.
+
+    By default, articles are created in a "draft"/unpublished state, but this
+    can be overridden by individual posts by specifying `published: true` in
+    their frontmatter, if you prefer a "just do it" approach.
+
+    This uses the DEV API, which is currently in beta: https://docs.dev.to/api
+
+    The required API key can be generated for your account by following the steps
+    outlined here: https://docs.dev.to/api/#section/Authentication
+    """
+
+    action_log(f"Hello? Yes, this is {SILO_NAME}.")
+    results = {
+        'added': {post.path:_create(post, api_key) for post in posts if not silo_id_for(post, SILO_NAME)},
+        'modified': {post.path:_update(post, api_key) for post in posts if silo_id_for(post, SILO_NAME)}
+    }
+    action_log("The results are in:")
+    action_log(pprint.pformat(results))
+    return results
+
+### privates ###
+
+def _create(post, api_key=None):
+    """
+    Creates a new article for the given post on DEV.to and returns the silo ID
+    and URL of the newly created article.
+
+    This tries to create an **unpublished** draft. However, the 'published'
+    status can be overridden in the frontmatter of the post itself for a
+    "just do it" approach.
+
+    @see https://docs.dev.to/api/#operation/createArticle
+    """
+    assert api_key, "missing API key"
+    assert post, "missing post"
+    assert fronted(post).get('title'), "article is missing a title"
```

Comment:
as a reminder, `assert` will be ignored when Python is running with optimizations. assertions are meant for debug/dev environments.

Reply:
Good point 💯 Better to replace most/all of these `asserts` with actual input handling and raise value errors when they fail.

Reply:
Done in https://github.com/dabrady/syndicate/pull/1/commits/a7606e890570c849e77faa35c14f33c382c8d04b